dirty_cat/similarity_encoder.py

```diff
@@ -266,7 +266,9 @@ def fit(self, X, y=None):
         -------
         self
         """
-        X = self._check_X(X)
+
+        Xlist, n_samples, n_features = self._check_X(X)
```

Comment:
Hum, I think that we should try to work on different versions of scikit-learn. Hence we probably need a test on the version of scikit-learn.

You can find examples of such test here:
https://github.com/nilearn/nilearn/search?q=looseversion&unscoped_q=looseversion